### ISSUE TITLE
The new input interface for the "Subject" and "Language" fields is awesome. Why not generalize it?

### ISSUE THREAD

**Issue by BPeuch** [IQSS/dataverse#7262]
Version: 4.20

Guten Tag,

We really like the new input interface for those fields:

![inpm](https://user-images.githubusercontent.com/15209224/92930159-fc3b0780-f441-11ea-8e0a-de8deee6700b.png)

Perhaps it could be generalized to all other fields that consist in a drop-drown menu with fixed values? E.g. "Geographic Coverage" > "Country / Nation."

One of our project partners, a social scientist, told us she was surprised when, having selected the dropped down menu for that same field, she started typing ```b e l``` in order to get "Belgium," but instead of receiving the letters as a unified sequence the software took them one at a time and eventually led her to "Lao People's Democratic Republic." This problem would be fixed with the new input interface.

**Comment by pdurbin** [comment]
@BPeuch good idea. I just wanted to point out that this interface isn't limited to those two fields. It's applied programatically based on features of the metadata field. It also applies to "Type" in the astro block, for example:

<img width="437" alt="Screen Shot 2020-09-11 at 11 47 20 AM" src="https://user-images.githubusercontent.com/21006/92946752-0e5a7d00-f425-11ea-9672-7d89f37b4312.png">

As you say, it's doesn't apply to "Country/Nation". Here's a screenshot.

<img width="737" alt="Screen Shot 2020-09-11 at 11 49 22 AM" src="https://user-images.githubusercontent.com/21006/92946754-0ef31380-f425-11ea-81a2-11a69bbd2a47.png">

TODO: put the pull request here where subject, language, etc. were changed to use the new interface.

**Comment by jggautier** [comment]
Since this GitHub issue was opened, the input interface for the "Subject" and "Language" fields has been generalized to all fields that are dropdown fields with a certain number of options, I think 9 or more or 10 or more.

> One of our project partners told us she was surprised when, having selected the drop-down menu for that same field, she started typing b e l in order to get "Belgium," but instead of receiving the letters as a unified sequence the software took them one at a time and eventually led her to "Lao People's Democratic Republic." This problem would be fixed with the new input interface.

It looks like this behavior was changed, too. When users type "b", the dropdown shows options with the letter "b" in them. Typing "be" shows options with the letters "be", in that order. At this point, the user doesn't see "Lao People's Democratic Republic", since "be" isn't in that option:

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/2c5dcbe9-6d3b-4297-9c48-eed5ce35efb0" />

I'm going to close this GitHub issue as completed.
